### PR TITLE
Add an alert that preview is not avaliable, for an empty test. Issue #6575

### PR DESCRIPTION
**Pull request by chtelmko**

### CODE REVIEW

**Comment by chtelmko** [comment]
I am having trouble recreating the behavior, where the preview button is showing up (even when there are no questions in the exam):
courses with empty exam has a preview

On my dev machine this looks like this (no preview button):
courses with empty exam  no preview

I believe this is due the ngif structural directive in the courses-view.component.html :
the ngif

I have added the alert "Preview is not available for this test" in the courses-view.component.ts. But I can't test the routing back without the preview button.

File: src/app/courses/step-view-courses/courses-step-view.component.html [hunk 1]
```diff
@@ -78,7 +78,7 @@ <h3 class="margin-lr-3 ellipsis-title"><ng-container i18n>Step</ng-container> {{
       (resourceUrl)="setResourceUrl($event)">
     </planet-resources-viewer>
   </div>
-  <ng-template #emptyRecord>
+  <ng-template #emptyRecord>
```
Comment (paulbert) [inline src/app/courses/step-view-courses/courses-step-view.component.html:81]: Please remove the extra spaces added in this line

File: src/app/courses/step-view-courses/courses-step-view.component.ts [hunk 2]
```diff
@@ -9,6 +9,7 @@ import { SubmissionsService } from '../../submissions/submissions.service';
 import { ResourcesService } from '../../resources/resources.service';
 import { DialogsSubmissionsComponent } from '../../shared/dialogs/dialogs-submissions.component';
 import { StateService } from '../../shared/state.service';
+import { PlanetMessageService } from '../../shared/planet-message.service';
```
Comment (paulbert) [inline src/app/courses/step-view-courses/courses-step-view.component.ts:12]: The changes in this file aren't doing anything.  Please remove.

File: src/app/exams/exams-view.component.ts [hunk 3]
```diff
@@ -9,13 +9,19 @@ import { CouchService } from '../shared/couchdb.service';
...
 import { Exam, ExamQuestion } from './exams.model';
+import { PlanetMessageService } from '../shared/planet-message.service';
 
 @Component({
   selector: 'planet-exams-view',
   templateUrl: './exams-view.component.html',
   styleUrls: [ './exams-view.scss' ]
 })
 
+// .mat-card {
```
Comment (paulbert) [inline src/app/exams/exams-view.component.ts:20]: Make sure you remove lines like this before taking off the **WIP** label.  Code that is commented out should not be merged into the master branch.

File: src/app/exams/exams-view.component.ts [hunk 4]
```diff
@@ -61,12 +68,12 @@ export class ExamsViewComponent implements OnInit, OnDestroy {
       this.previewMode = params.get('preview') === 'true' || this.isDialog;
       this.questionNum = +params.get('questionNum') || this.questionNum;
       if (this.previewMode) {
-        ((this.exam || this.submission) ? of({}) : this.couchService.get(`exams/${params.get('examId')}`)).subscribe((res) => {
+  ((this.exam || this.submission) ? of({}) : this.couchService.get(`exams/${params.get('examId')}`)).subscribe((res) => {
```
Comment (paulbert) [inline src/app/exams/exams-view.component.ts:71]: Please be more careful when making changes.  This line and line 76 should not have the indentation changed.

**Pull request merged**